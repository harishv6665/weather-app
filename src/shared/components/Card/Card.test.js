import React from "react";
import { shallow, mount } from "enzyme";
import {Card} from "../";

const CARD_CONTENT_TEXT = "Card content here";

const CardContent = () => <span>{CARD_CONTENT_TEXT}</span>

describe("Card tests", () => {

    it("should be defined", () => {
        expect(Card).toBeDefined();
    });

    it("should match to snapshot", () => {
        const wrapper = shallow(<Card>
            <CardContent />
        </Card>);
        expect(wrapper).toMatchSnapshot();
    });

    it("should match the DOM structure", () => {
        const wrapper = mount(<Card>
            <CardContent />
        </Card>);
        expect(wrapper.find("div").length).toBe(1);
        expect(wrapper.find("span").length).toBe(1);
        expect(wrapper.find("div span").length).toBe(1);
        expect(wrapper.find("div span").text()).toEqual(CARD_CONTENT_TEXT);
    });

});
