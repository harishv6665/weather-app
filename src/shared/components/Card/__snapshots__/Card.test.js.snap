// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card tests should match to snapshot 1`] = `ShallowWrapper {}`;
